src/pyOpenMS/addons/MatrixDouble.pyx

```diff
@@ -3,23 +3,31 @@ from Matrix cimport *
 
 # continue with extra code if needed
 
-    def get_matrix(self):
+    def get_matrix_as_view(self):
         """Cython signature: numpy_matrix get_matrix()
         """
 
         cdef _Matrix[double] * mat_ = self.inst.get()
 
         cdef unsigned int rows = mat_.rows()
         cdef unsigned int cols = mat_.cols()
+        cdef unsigned int n = rows * cols
+        cdef np.ndarray[double, ndim=2] data
+        data = np.zeros( (rows,cols), dtype=np.float64)
 
-        cdef libcpp_vector[double] tmp_vec;
-        tmp_vec = mat_.asVector();
+        cdef libcpp_vector[double] * vec_ptr = <libcpp_vector[double]*> mat_
```

Comment:
@jpfeuffer k got nearly all tests working. except the ones that test the correct behavior of the get matrix as view (at least it looks like it) 
Can this even work??? Because I think I would need to call getEigenMatrix().data() and cast that one. But getEigenMatrix() is not wrapped... like this it seems to cast the storage of the Matrix class in memory to a vector.
what should I do here? I could of course expose a double* or similr in the cpp class but is there something more elegant?

Reply:
No, you have to expose some kind of access to the underlying buffer.
You can store a shared ptr in the matrix class, make the Eigen Matrix a Map/view on this and hand out references to this shared ptr if you are worried about memory safety.

Reply:
Or get rid of the wrapper around Eigen Matrix completely.

Reply:
k had a quick test with
```
from Types cimport *
from libcpp cimport bool
from libcpp.pair cimport pair as libcpp_pair
from libcpp.vector cimport vector as libcpp_vector

cdef extern from "<OpenMS/DATASTRUCTURES/Matrix.h>" namespace "OpenMS":
    
    cdef cppclass Matrix[ValueT]:
        # wrap-instances:
        #  MatrixDouble := Matrix[double]

        Matrix() except + nogil 
        Matrix(Matrix[ValueT]) except + nogil 
        Matrix(size_t rows, size_t cols, ValueT value) except + nogil 
        ValueT getValue(size_t i, size_t j) nogil
        void setValue(size_t i, size_t j, ValueT value) nogil
        size_t rows() nogil
        size_t cols() nogil
        size_t size() nogil
        ValueT* data()        <---------------------- added this
```
I commented out the pyx code but now I get an error in ConversionProvider

NameError: no converter for double * in: defaultdict(<class 'list'>, ...

Reply:
ah probably forgot the cdef

Reply:
hmm still same error. any idea?

Reply:
you will not be able to wrap that into Python.

use #no-wrap or how it was called.